commerce_coordinator/apps/rollout/pipeline.py

```diff
@@ -31,17 +34,29 @@ def run_filter(self, request):  # pylint: disable=arguments-differ
         """
         Execute a filter with the signature specified.
         Arguments:
-            request: request object passed through from the lms filter
+            request: request object passed through from the LMS filter
         Returns:
             dict:
                 active_order_management_system: result from pipeline steps to determine if
-                            redirect_to_commercetools_checkout flag is enabled and course_run_key
-                            and sku query params exist and detect which checkout to redirect to
+                            redirect_to_commercetools_checkout flag is enabled and course_run_key,
+                            sku, or bundle query params exist, and detect which checkout to redirect to.
         """
         sku = request.query_params.get('sku', '').strip()
         course_run = request.query_params.get('course_run_key', '').strip()
+        bundle = request.query_params.get('bundle', '').strip()
         commercetools_available_course = None
 
+        ct_api_client = CommercetoolsAPIClient()
+
+        if bundle and is_program_redirection_to_ct_enabled(request):
+            commercetools_available_program = ct_api_client.get_product_variant_by_program(bundle)
+            if commercetools_available_program:
+                return {ACTIVE_ORDER_MANAGEMENT_SYSTEM_KEY: COMMERCETOOLS_FRONTEND}
+            logger.warning(
+                f'[get_product_variant_by_program] Program {bundle} not found in Commercetools. '
+                f'Please ensure it is properly synced.'
+            )
```

Comment:
We need to emit the log (and configure an alert on it) for program requests in the following cases (all leading to the legacy redirection):
- If the CT API endpoint responds with a non-200 exception
- If no product is found in the CT product search